### PR TITLE
examples: update structure

### PR DESCRIPTION
**Pull request by robert-zaremba**
I've organized and renamed examples and prepared a new example for the reference templates.

### CODE REVIEW

**Review by ratik21 — COMMENTED** [review]
`examples/asa` contains scripts for Algorand Smart Contracts as well (contract and delegated). We should either rename this folder (`/asa`) or move the smart contract scripts to (`/asc`). What do you think?

File: README.md [hunk 1]
```diff
@@ -22,6 +22,17 @@ User documentation is available in [/docs](docs/README.md).
...
 
+### Examples
+
+Our `/examples` directory provides few projects with smart contracts and ASA:
+
++ Please start with reading Algorand reference documentation about [smart contract](https://developer.algorand.org/docs/reference/teal/specification/).
++ Don't forget to study Algorand smart contract [guidelines](https://developer.algorand.org/docs/reference/teal/guidelines/).
++ Go to the [examples/ref-templates](./examples/ref-templates/README.md) to see how the reference templates are implemented.
++ Then go to [examples/asa](./examples/ref-templates/README.md) to learn how you can easily manage and deploy ASA with `algob`.
++ check other examples as well.
```
Comment (ratik21) [inline README.md:33]:
```suggestion
+ Check other examples as well.
```

**Comment by robert-zaremba** [comment]
> `examples/asa` contains scripts for Algorand Smart Contracts as well (contract and delegated). We should either rename this folder (`/asa`) or move the smart contract scripts to (`/asc`). What do you think?

With the new task, we want that user will start with `ref-templates` task first. It provides a first steps with smart contracts. So, we can keep the `asa` name - because this one will be first steps with ASA + how to use it with smart contracts.

**Comment by ratik21** [comment]
> With the new task, we want that user will start with `ref-templates` task first. It provides a first steps with smart contracts. So, we can keep the `asa` name - because this one will be first steps with ASA + how to use it with smart contracts.
:+1:

File: examples/asa/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# ASA deployment using [Algorand Builder](https://github.com/scale-it/algorand-builder/)
+
+This project shows how to create Algorand Standard Asset (ASA).
+We will create gold and tesla assets. In algob, we recommend to
+create assets using a specification file, as provided in `assets/asa.yaml`.
+
+Each of those tokens are owned by their specific owner accounts.
```
Comment (amityadav0) [inline examples/asa/README.md:7]:
```suggestion
Each of these assets are owned by their specific owner accounts.
```

File: examples/asa/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+After the creation a third account is opted-in in order to receive the tokens.
```
Comment (amityadav0) [inline examples/asa/README.md:8]:
```suggestion
After the creation, a third account is opted-in in order to receive the assets.
```

**Review by ratik21 — APPROVED** [review]

**Pull request merged**